coffee_core/src/coffee.rs

```diff
@@ -234,6 +234,28 @@ impl PluginManager for CoffeeManager {
         Err(err)
     }
 
+    async fn remove(&mut self, plugin: &str) -> Result<Value, CoffeeError> {
+        debug!("removing plugin: {plugin}");
+        let values: &mut Vec<Plugin> = &mut self.config.plugins;
+        if let Some(index) = values.iter().position(|x| x.name() == plugin) {
+            let plugin = values[index].clone();
+            let exec_path = plugin.exec_path.clone();
+            debug!("runnable plugin path: {exec_path}");
+            values.remove(index);
+            self.coffe_cln_config
+                .rm_conf("plugin", Some(&exec_path.to_owned()))
+                .map_err(|err| CoffeeError::new(1, &err.cause))?;
+            self.storage.store(&self.storage_info()).await?;
+            self.update_conf().await?;
+            return Ok(serde_json::to_value(plugin).unwrap());
+        } else {
+            return Err(CoffeeError::new(
+                1,
+                &format!("plugin `{plugin}` is already not installed"),
+            ));
+        }
```

Comment:
Can we return a typed response? as I did in some previous method like `coffee_list` and `coffee_list_remotes`?

Reply:
Yes i created a new struct `CoffeeRemove` with one field `plugin`